src/Microsoft.Health.Dicom.Core/Features/CustomTag/IndexableDicomTagService.cs

```diff
@@ -0,0 +1,68 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Dicom;
+using EnsureThat;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Dicom.Core.Configs;
+using Microsoft.Health.Dicom.Core.Extensions;
+using Microsoft.Health.Dicom.Core.Features.Query;
+
+namespace Microsoft.Health.Dicom.Core.Features.CustomTag
+{
+    public class IndexableDicomTagService : IIndexableDicomTagService
+    {
+        private readonly ICustomTagStore _customTagStore;
+        private readonly bool _enableCustomQueryTags;
+        public static readonly IReadOnlyList<IndexableDicomTag> CoreIndexableDicomTags = GetCoreIndexableDicomTags();
```

Comment:
Do the initializations as late as possible. Maybe in the constructor

Reply:
I think there are multiple ways to initialize:
1. static field + initialize in field
Pros:  don't need to worry about thread safe
Cons:  add application start time.
 2. static field + initialize in static constructor
Got build failure from code analyzer
3. instance field + initialize in constructor
Pros: only initialized when need.
Cons: need to initialize for every request and handle with thread-safe.

My preference is the No.1, since this is very small code, take nanosecond level execution time, should be acceptable.